app/javascript/packages/document-capture/context/upload.tsx

```diff
@@ -0,0 +1,148 @@
+import { createContext } from 'react';
+import { useObjectMemo } from '@18f/identity-react-hooks';
+import defaultUpload from '../services/upload';
+
+const UploadContext = createContext({
+  upload: defaultUpload,
+  getStatus: () => Promise.reject() as () => Promise<UploadSuccessResponse>,
+  statusPollInterval: undefined as number,
+  isMockClient: false,
+  backgroundUploadURLs: {} as Record<string, string>,
+  backgroundUploadEncryptKey: undefined as CryptoKey,
+  flowPath: 'standard' as FlowPath,
+  csrf: null as string,
+});
+
+UploadContext.displayName = 'UploadContext';
+
+type ReactNode = import('react').ReactNode;
+
+type FlowPath = 'standard' | 'hybrid';
+
+// Upload field error, after normalized to error instance.
+export interface UploadFieldError {
+  // Field name
+  field: 'front' | 'back' | 'selfie' | 'network';
+
+  // Error message.
+  message: string;
+}
+
+interface UploadOptions {
+  // HTTP method to send payload.
+  method?: 'POST' | 'PUT';
+
+  // Endpoint to which payload should be sent.
+  endpoint: string;
+
+  // CSRF token to send as parameter to upload implementation.
+  csrf: string;
+}
+
+export interface UploadSuccessResponse {
+  // Whether request was successful.
+  success: true;
+
+  // Whether verification result is still pending.
+  isPending: boolean;
+}
+
+export interface UploadErrorResponse {
+  // Whether request was successful.
+  success: false;
+
+  // Error messages.
+  errors: UploadFieldError[];
+
+  // URL to which user should be redirected.
+  redirect: string;
+
+  // Number of remaining doc capture attempts for user.
+  remaining_attempts: number;
+
+  // Boolean to decide if capture hints should be shown with error.
+  hints: boolean;
+}
+
+export type UploadImplementation = (
+  payload: Record<string, any>,
+  options: UploadOptions,
+) => Promise<UploadSuccessResponse>;
+
+interface UploadContextProviderProps {
+  // upload Custom upload implementation.
```

Comment:
I think "upload" snuck in here.

```suggestion
  /**
   * Custom upload implementation.
   */
```